src/main/java/org/venice/beachfront/bfapi/auth/jwt/JWTAuthProvider.java

```diff
@@ -0,0 +1,127 @@
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.jwt.Jwt;
+import org.springframework.stereotype.Component;
+import org.venice.beachfront.bfapi.auth.BfAuthenticationToken;
+import org.venice.beachfront.bfapi.model.UserProfile;
+import org.venice.beachfront.bfapi.services.UserProfileService;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import model.logger.Severity;
+import util.GeoAxisJWTUtility;
+import util.PiazzaLogger;
+
+/**
+ * Handles the authentication and authorization for all REST Requests that define Bearer Authentication, in the form of
+ * JWTs. This will validate and verify the JWT Tokens and associate the JWT request with the User Profile that contains
+ * the Distinguished Name as stated in the JWT token.
+ * <p>
+ * In present form, JWT tokens will only be valid for users who have already authenticated via OAuth, and as such
+ * already have a UserProfile in the Beachfront system.
+ */
+@Component
+public class JWTAuthProvider implements AuthenticationProvider {
+	@Autowired
+	private UserProfileService userProfileService;
+	@Autowired
+	private PiazzaLogger piazzaLogger;
+	@Autowired
+	private GeoAxisJWTUtility jwtUtility;
+	@Autowired
+	private ObjectMapper objectMapper;
+
+	@Value("${jwt.enabled}")
+	private Boolean jwtEnabled;
+
+	@Override
+	public Authentication authenticate(Authentication authentication) throws AuthenticationException {
+		// Ensure JWT is Enabled
+		if (!jwtEnabled.booleanValue()) {
+			throw new JWTAuthenticationException("JWT authentication is not enabled.");
+		}
+
+		// Check for the validity of the JWT
+		String encodedJwt = authentication.getPrincipal().toString();
+		Jwt jwt = jwtUtility.decodeAndVerify(encodedJwt);
+		if (jwt == null) {
+			String error = "JWT received from the client that could not be verified.";
+			piazzaLogger.log(error, Severity.INFORMATIONAL);
+			throw new JWTAuthenticationException(error);
+		}
+
+		// Get the DN from the JWT Payload and retrieve the User Profile
+		String dn = null;
+		JsonNode payloadJson = null;
+		try {
+			String jwtPayload = jwt.getClaims();
+			payloadJson = objectMapper.readTree(jwtPayload);
+			JsonNode dnNode = payloadJson.get("dn");
+			if (dnNode == null) {
+				throw new IOException("Distinguished Name from JWT Payload is not provided.");
+			}
+			dn = payloadJson.get("dn").asText();
+			if ((dn == null) || (dn == "")) {
+				throw new IOException("Distinguished Name from JWT Payload is empty or blank.");
+			}
+		} catch (IOException exception) {
+			String error = String.format("Valid JWT received from the client, but could not read DN from Payload with error: %s.",
+					exception.getMessage());
+			piazzaLogger.log(error, Severity.ERROR);
+			throw new JWTAuthenticationException(error);
+		}
+
+		// Check the expiration
+		long expirationEpoch = payloadJson.get("exp").asLong();
```

Comment:
Consider adding `Secs` to the end of this variable name for clarity on the units that it's holding.

Reply:
Great, fixed